bcp47/library/Data/BCP47/Trie/Internal.hs

```diff
@@ -66,6 +68,13 @@ unionWith f = unionUsing (liftA2 f)
 unionUsing :: (Maybe a -> Maybe a -> Maybe a) -> Trie a -> Trie a -> Trie a
 unionUsing f (Trie x) (Trie y) = Trie $ Map.unionWith (union2Using f) x y
 
+nullToMaybe :: Map k a -> Maybe (Map k a)
+nullToMaybe m = if Map.null m then Nothing else Just m
+
+-- Like `Map.mapMaybe` but returns a `Maybe` because `Trie` should be non-empty
+mapMaybe :: (a -> Maybe b) -> Trie a -> Maybe (Trie b)
+mapMaybe f (Trie x) = Trie <$> nullToMaybe (Map.mapMaybe (mapMaybe2 f) x)
```

Comment:
It would be nice to get a quick check invariant for this.

Reply:
Good idea- I added one for making sure that we have the same number of `Just`s before and after